From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joshua Peraza <jperaza@google.com>
Date: Tue, 1 Sep 2026 11:56:07 -0700
Subject: [m152] Round up OOB stack pointers in ScanStackForPointers

Bug: 502304489
Change-Id: I95ce1aa997e52aad13b06f2412c68ce361ee4f82
Fixed: 553316023
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8332269
Auto-Submit: Joshua Peraza <jperaza@chromium.org>
Reviewed-by: Mark Mentovai <mark@chromium.org>
Commit-Queue: Joshua Peraza <jperaza@chromium.org>
Cr-Commit-Position: refs/branch-heads/7977@{#2267}
Cr-Branched-From: b7fe14017379ddffae396d944fb8b59a5896c261-refs/heads/main@{#1669021}
(cherry picked from commit 4d2a8274815fd990129b3dd06f9fd74bfddb9297)

diff --git a/third_party/crashpad/README.chromium b/third_party/crashpad/README.chromium
index 4f8960d84ff7f15b23a719c6f33d0dfa29a20e3f..628989ed60954799df5bdd804999bf6daf963f43 100644
--- a/third_party/crashpad/README.chromium
+++ b/third_party/crashpad/README.chromium
@@ -60,3 +60,4 @@ Local Modifications:
    "base/allocator/partition_allocator/src/partition_alloc/partition_alloc_buildflags.h",
    https://crbug.com/1467773
  - CPEPrefix for third_party/zlib is added to README.chromium
+ - 60dd943f48d7 Round up OOB stack pointers in ScanStackForPointers
diff --git a/third_party/crashpad/crashpad/snapshot/sanitized/process_snapshot_sanitized.cc b/third_party/crashpad/crashpad/snapshot/sanitized/process_snapshot_sanitized.cc
index afa1c9f849351dee309413331d48a802f57fd570..5d44229888984d630b683db201a6cabd7eb1fffd 100644
--- a/third_party/crashpad/crashpad/snapshot/sanitized/process_snapshot_sanitized.cc
+++ b/third_party/crashpad/crashpad/snapshot/sanitized/process_snapshot_sanitized.cc
@@ -52,7 +52,11 @@ class StackReferencesAddressRange : public MemorySnapshot::Delegate {
   template <typename Pointer>
   bool ScanStackForPointers(void* data, size_t size) {
     size_t sp_offset;
-    if (!AssignIfInRange(&sp_offset, stack_pointer_ - stack_->Address())) {
+    // stack_pointer_ might point below the stack region if ProcessReaderLinux
+    // interpreted it as pointing to a guard page.
+    if (stack_pointer_ < stack_->Address()) {
+      sp_offset = 0;
+    } else if (!AssignIfInRange(&sp_offset, stack_pointer_ - stack_->Address())) {
       return false;
     }
     const size_t aligned_sp_offset =
